libraries/chain/database.cpp

```diff
@@ -3836,38 +3843,66 @@ void database::update_last_irreversible_block()
       }
    }
 
-   commit( dpo.last_irreversible_block_num );
-
    for( uint32_t i = old_last_irreversible; i <= dpo.last_irreversible_block_num; ++i )
```

Comment:
When i= old_last_irreversible or last_irreversible_block_num, it will be notified twice.